arrows/ocv/resection_camera.cxx

```diff
@@ -101,6 +140,21 @@ ::check_configuration( vital::config_block_sptr config ) const
                ", needs to be greater than zero." );
     good_conf = false;
   }
+
+  std::stringstream ss( config->get_value< std::string >( "focal_scales",
+                                                          "1" ) );
+  vectorf focal_scales;
+  ss >> focal_scales;
+
+  auto m = std::min_element( focal_scales.begin(), focal_scales.end() );
+  if( *m <= 0 )
```

Comment:
If `focal_scales` is empty then then this will dereference `focal_scales.end()`, which will likely crash.  You should check if `focal_scales` is empty.  It should be a different error message if the `focal_scales` are empty.  We need at least one positive value unless you are going to treat `1` as a special value that is always implicitly in the list.  However, currently that is not the case that 1 is always forced to be in the list.

Reply:
I think the prior line
std::stringstream ss( config->get_value< std::string >( "focal_scales", "1" ) );
ensures that we always have at least one value for focal_scales, so it is never empty; should I leave a comment about that in the code?

Reply:
Not exactly.  That "1" provides a default value if the option is unspecified.  If the option is specified, but is left blank then you will still get an empty string.  Or if the values specified are not a valid list of number the stringstream may give you an empty vector.

Reply:
now requiring that focal_scales is non-empty